### ISSUE TITLE
[data-grid] Remove prop-types warning when passing an array to a cell

### ISSUE THREAD

**Issue by erfanansari** [mui/mui-x#4109]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

[DataGrid] it looks like Datagrid dropped support for rendering arrays as grid cells, I could not find this change on the changelog
is it a bug?
in the previous versions (before 5.6.0), this wasn't a problem, but after updating to 5.6.0 it shows a warning

### Steps to reproduce 🕹

Steps:

1.
2.
3.
4.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by flaviendelangle** [comment]
Hi,
Thanks for your contribution.

Could you provide a working reproduction, this [Codesandbox template](https://codesandbox.io/s/datagrid-v5-quick-start-54iz1) might be a good starting point.

**Comment by erfanansari** [comment]
Hi there,
Here is the [link](https://codesandbox.io/s/datagrid-v5-quick-start-forked-97gxes?file=/src/App.tsx)

**Comment by flaviendelangle** [comment]
Indeed...
It is just a proptypes warning so if your build process removes proptypes, you won't have it on production.
But I will fix it in the next release :+1:

**Comment by flaviendelangle** [comment]
@m4theushw are you okay to use `any` instead of `GridCellValue` ?
Nothing prevents the non-formatted value of a cell to be an array, or any other type.

**Comment by m4theushw** [comment]
I don't know why `GridCellValue` is used instead of `any`. This same question was already raised in https://github.com/mui/mui-x/issues/1329#issuecomment-814478259. We can try using `any`.

**Comment by peter-oms** [comment]
Hi @flaviendelangle, thanks a lot of resolving this! 🙂 I see it's been merged 2 days ago, but the latest 5.7.0 release is from 7 days ago, so this is not released yet as far as I can tell.

Is there a way I can easily track whether some merged branch has been released and/or track the future releases timeline? Would be great if these are tagged (against milestones maybe) or have some other way to easily spot when a functionality has been/will be released.

Thanks a lot again - looking forward to trying this out in action once it's been released (in 5.7.1 I hope) 🙌

**Comment by flaviendelangle** [comment]
The best way it to check if your PR number appears in [the CHANGELOG.md](https://github.com/mui/mui-x/blob/master/CHANGELOG.md) file

The milestones concept is great, but we struggle to use it correctly on Github.
